Benemérita Universidad Autónoma de Puebla
![image](https://drive.google.com/uc?export=view&id=1Ugk24FB0FcMahW9jSEdIRPSZUoOUYnGy)



**<font color=#3333ff>Facultad de Ciencias Físco Matemáticas</font>**

![image 2](https://drive.google.com/uc?export=view&id=1C25kNo7qgW2tlW1SOqpmlyfnaZAG5IvJ)

> Posgrado en Ciencias (Física Aplicada)

Métodos Matemáticos

Prof. Dr. Javier M. Hernández Lopez

T10, P1
Resolución de integral beta, con método númerico

Estudiante: 

*   Angel Ricardo Sánchez Zeferino

In [ ]:
#  Libraries 
%pylab inline
from scipy import integrate

Populating the interactive namespace from numpy and matplotlib


In [ ]:
# Esta función la desarrolle en el trabajo de exposición
def simpson_function(f, a, b, N, args=None): 
  """
  This function implements the Simpson´s rule, giving by eq. (4),  needs four inputs:
  f: The function that we´ll integrate
  a,b: The start point and the end point of interval of integration [a,b]
  N: The number of equal segments of width h
  arg: list of arguments for the function
  """
  if args is None:
   args = []                        # If the arguments of the function is none, then create a list empty
  x=linspace(a,b,2*N)               # Create a vector  with 2N items between a and b
  h=(b-a)/(2*N-1)                   # Crete the length of segments, for this substract the start point and the start point
                                    # and then divided by space for 2N elements
  y=f(x)            # Apply the function to x (in form of list)
                                    # If the number of args is unknown, add * before the parameter name
                                    # Slice function in list: Notation [start:stop:step]
  evens = y[2:-2:2]                 # Take the items from index=2 to index = -2 in steps of 3 items 
  odds= y[1:-1:2]                   #
  return (y[0] + 2*odds.sum()+ 4*evens.sum() + y[-1] )*h/3.0  # Do the sum 

In [ ]:
simpson_function(lambda x : ((sin(x))**6)*((cos(x))**4),0,pi/2,10000)

0.01840776945462769

In [ ]:
# Se comprueba con cuadratura de Clenshaw-Curtis
quadResult = integrate.quad(lambda x : ((sin(x))**6)*((cos(x))**4), 0, pi/2)
quadResult

El resultado analítico fue

$
(45 \pi)/(64 \times 5!) = 0.0184
$

Por lo tanto el resultado es el mismo.